src/Map.ts

```diff
@@ -518,111 +526,113 @@ export class Map extends maplibregl.Map {
         // No tiles.json found (should not happen on maintained styles)
       }
 
+      // Check if default controls are enabled (using the default value)
+      const defaultControlsEnabled = options.defaultControls;
+
       // The attribution and logo must show when required
       if (options.forceNoAttributionControl !== true) {
         if ("logo" in tileJsonContent && tileJsonContent.logo) {
           const logoURL: string = tileJsonContent.logo;
 
           this.addControl(
-            new MaptilerLogoControl({ logoURL }),
+            new MaptilerLogoControl({ defaultControlsEnabled, logoURL }),
             options.logoPosition,
           );
         } else if (options.maptilerLogo) {
-          this.addControl(new MaptilerLogoControl(), options.logoPosition);
+          this.addControl(
+            new MaptilerLogoControl({ defaultControlsEnabled }),
+            options.logoPosition,
+          );
         }
       }
 
-      // the other controls at init time but be after
-      // (due to the async nature of logo control)
-
-      // By default, no scale control
-      if (options.scaleControl) {
-        // default position, if not provided, is top left corner
-        const position = (
-          options.scaleControl === true || options.scaleControl === undefined
-            ? "bottom-right"
-            : options.scaleControl
-        ) as ControlPosition;
-
-        const scaleControl = new ScaleControl({ unit: config.unit });
-        this.addControl(scaleControl, position);
-        config.on("unit", (unit) => {
-          scaleControl.setUnit(unit);
-        });
+      if (!defaultControlsEnabled) {
+        this._controlContainer.style.display = "none";
+        DOMremove(this._controlContainer);
       }
+      // Only add default controls if defaultControls is not false
+      if (defaultControlsEnabled) {
+        // By default, no scale control
+        if (options.scaleControl) {
```

Comment:
should this be `options.scaleControl !== false` like the others? or is scaleControl an exception?